pkg/client/namespaced_client_test.go

```diff
@@ -0,0 +1,597 @@
+			deleteDeployment(ctx, dep, ns)
+		})
+
+		It("should successfully List objects when namespace is not specified with the object", func() {
+			result := &appsv1.DeploymentList{}
+			opts := client.MatchingLabels(dep.Labels)
+
+			Expect(getClient().List(ctx, result, opts)).NotTo(HaveOccurred())
+			Expect(len(result.Items)).To(BeEquivalentTo(1))
+			Expect(result.Items[0]).To(BeEquivalentTo(*dep))
+		})
+
+		It("should List objects from the namespace specified in the client", func() {
+			result := &appsv1.DeploymentList{}
+			opts := client.InNamespace("non-default")
+
+			Expect(getClient().List(ctx, result, opts)).NotTo(HaveOccurred())
+			Expect(len(result.Items)).To(BeEquivalentTo(1))
+			Expect(result.Items[0]).To(BeEquivalentTo(*dep))
+		})
+	})
+
+	Describe("Create", func() {
+		AfterEach(func() {
+			deleteDeployment(ctx, dep, ns)
+		})
+
+		It("should successfully create object in the right namespace", func() {
+			By("creating the object initially")
+			err := getClient().Create(ctx, dep)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("checking if the object was created in the right namespace")
+			res, err := clientset.AppsV1().Deployments(ns).Get(ctx, dep.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(res.GetNamespace()).To(BeEquivalentTo(ns))
+		})
+
+		It("should not create object if the namespace of the object is different", func() {
+			By("creating the object initially")
+			dep.SetNamespace("non-default")
+			err := getClient().Create(ctx, dep)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should create an object in the namespace specified with the client", func() {
+			By("creating the object initially")
+
+			err := getClient().Create(ctx, dep)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("checking if the object was created in the right namespace")
+			res, err := clientset.AppsV1().Deployments(ns).Get(ctx, dep.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(res.GetNamespace()).To(BeEquivalentTo(ns))
+		})
+		It("should create a cluster scoped object", func() {
+			cr := &rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   fmt.Sprintf("clusterRole-%v", count),
+					Labels: map[string]string{"name": fmt.Sprintf("clusterRole-%v", count)},
+				},
+			}
+
+			cr.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   "rbac.authorization.k8s.io",
+				Version: "v1",
+				Kind:    "ClusterRole",
+			})
+
+			By("creating the object initially")
+			err := getClient().Create(ctx, cr)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("checking if the object was created")
+			res, err := clientset.RbacV1().ClusterRoles().Get(ctx, cr.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(res).NotTo(BeNil())
+
+			// Delete the clusterRole Resource
+			deleteClusterRole(ctx, cr)
+		})
+	})
+
+	Describe("Update", func() {
+		var err error
+		BeforeEach(func() {
+			dep.Annotations = map[string]string{"foo": "bar"}
```

Comment:
I think you meant to add that after the creation, all the `Update` calls in this test are no-ops